svm/src/transaction_processor.rs

```diff
@@ -484,32 +469,65 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         let fee_payer_index = 0;
         validate_fee_payer(
             fee_payer_address,
-            &mut fee_payer_account,
+            &mut loaded_fee_payer.account,
             fee_payer_index,
             error_counters,
             rent_collector,
             fee_details.total_fee(),
         )?;
 
+        // If the nonce has been used in this batch already, we must drop the transaction
+        // This is the same as if it was used in different batches in the same slot
+        // If the nonce account was closed in the batch, we error as if the blockhash didn't validate
+        // We must vaidate the account in case it was reopened, either as a normal system account, or a fake nonce account
+        if let Some(ref advanced_nonce_info) = advanced_nonce {
```

Comment:
this block seems like it may benefit from being its' own function. That may make it much simpler to just test all the branches?

Reply:
youre right, i made it its own function. incidentally, do you think the caller should pass through the blockhash instead of the durable nonce, and the nonce validation function should hash it? the current way, every batch does a sha256sum once, the other way we would do it once per nonce transaction, but presumably most batches dont have any nonce transactions at all

asking first since changing the signature of `validate_transaction_fee_payer` changes a lot of tests so i would want to squash the history back to one code and n test commits. incidentally i also noticed that function doesnt need `&self` anymore so ill remove that too

Reply:
[2364af1](https://github.com/anza-xyz/agave/pull/3146/commits/2364af1431ffe91ba6397824eee6ee74986d0619)